spring-boot-samples/spring-boot-sample-data-geode/src/test/java/sample/data/geode/SampleDataGeodeApplicationTest.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2010-2013 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package sample.data.geode;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.data.gemfire.GemfireTemplate;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+
+import sample.data.geode.service.Calculator;
+
+/**
+ * The SampleDataGeodeApplicationTest class is a test suite of test cases testing the functionality
+ * of the {@link SampleDataGeodeApplication} when performing mathematical calculations.
+ *
+ * @author John Blum
+ * @see org.junit.Test
+ * @see org.junit.runner.RunWith
+ * @see org.springframework.data.gemfire.GemfireTemplate
+ * @see org.springframework.test.context.ContextConfiguration
+ * @see org.springframework.test.context.junit4.SpringJUnit4ClassRunner
+ * @see sample.data.geode.SampleDataGeodeApplication
+ * @since 1.4.0
+ */
+@RunWith(SpringJUnit4ClassRunner.class)
+@ContextConfiguration(classes = SampleDataGeodeApplication.class)
+@SuppressWarnings("unused")
+public class SampleDataGeodeApplicationTest {
+
+	@Autowired
+	@Qualifier("calculatorResolver")
+	private Calculator<Long, Long> calculator;
+
+	@Autowired
+	private GemfireTemplate calculationsTemplate;
+
+	@Value("${sample.data.geode.calculator:factorial}")
```

Comment:
I prefer not to use a specific properties file(s).  GemFire/Geode also contains some magic for "discovering" properties files and if the user start mixing and matching application properties with GemFire/Geode System properties, they are sure to get themselves in trouble since GemFire/Geode pukes on any non-recognized property.

The idea behind using the "custom, sample" application-specific properties was to allow the user to override them if they choose (e.g. `sample.data.geode.log.level` to affect the log output in Geode (defaulting to "config"), or the `sample.data.geode.calculator` to affect the calculation performed (defaulting to "factorial")) using reasonable defaults if the user does not specify anything.  I.e. no extraneous configuration is necessary, but the sample can be adjusted if the user desires.

Perhaps, I am just not as familiar as I should be with all of _Spring Boot's_ (application) configuration techniques and there is possibly a better/recommended way to do what I did?

I am also happy to remove these custom, sample application-specific properties if so desired.  Just say the word.